incydr/cli/cmds/alerts.py

```diff
@@ -286,3 +323,44 @@ def _create_query(**kwargs):
                 continue
             query = query.equals(field_option_map[k], v)
     return query
+
+
+def _get_cursor_store(api_key):
+    """
+    Get cursor store for alerts search checkpoints.
+    """
+    dir_path = get_user_project_path(
+        api_key,
+        "alert_checkpoints",
+    )
+    return BaseCursorStore(dir_path, "alerts")
+
+
+def _update_checkpoint(cursor, checkpoint_name, alerts_gen):
+    """
+    De-duplicates events across checkpointed runs.
+
+    Since using the timestamp of the last event
+    processed as the `--start` time of the next run causes the last event to show up again in the
+    next results, we hash the last event(s) of each run and store those hashes in the cursor to
+    filter out on the next run.
+
+    It's also possible that two events have the exact same timestamp, so
+    `checkpoint_events` needs to be a list of hashes so we can filter out everything that's actually
+    been processed.
+    """
+    # TODO: do we need to hash these?
```

Comment:
Used alert IDs to dedupe alert results instead of hashed alerts.  Do we need to hash alert IDs?

Reply:
We shouldn't need to hash. That was only necessary for audit log events, as they don't have a unique id.